lib/npm/test.js

```diff
@@ -56,6 +56,26 @@ function test(context, next) {
     }
     const proc = spawn(nodeBin, [npmBin, 'test', '--loglevel=' +
         context.options.npmLevel], options);
+
+    // Default timeout to 10 minutes if not provided
+    let delay = context.options.timeoutLength || 1000 * 60 * 10;
+    const timeout = setTimeout(endTest, delay);
+
+    function endTest(err, ret) {
+      if (endTest.run) throw new Error('insane state, process finished twice');
+      endTest.run = true;
+
```

Comment:
Not a bit fan of appending information to a function like this.

We already have timeout logic in npm.install

Is there a way to either abstract it out or replicate current timeout logic?